app/gui/src/dashboard/services/RemoteBackend.ts

```diff
@@ -1332,10 +1332,40 @@ export default class RemoteBackend extends Backend {
     }
   }
 
-  /** Download from an arbitrary URL that is assumed to originate from this backend. */
-  override async download(url: string, name?: string) {
-    download.download(url, name)
-    return Promise.resolve()
+  /** Download an asset. */
+  override async download(id: backend.AssetId, title: string) {
+    const asset = backend.extractTypeFromId(id)
+    switch (asset.type) {
+      case backend.AssetType.project: {
+        const details = await this.getProjectDetails(asset.id, true)
+        invariant(details.url != null, 'The download URL of the project must be present.')
+        download.download(details.url, `${title}.enso-project`)
+        break
+      }
+      case backend.AssetType.file: {
+        const details = await this.getFileDetails(asset.id, title, true)
+        invariant(details.url != null, 'The download URL of the file must be present.')
```

Comment:
I think here would be better to throw errors/reject promise instead. 
See: "You're fine throwing a descriptive error here because it's just very unlikely this will ever happen and even if it does you wouldn't really know what to do about it anyway."

https://www.npmjs.com/package/@epic-web/invariant#:~:text=You%27re%20fine%20throwing%20a%20descriptive%20error%20here%20because%20it%27s%20just%20very%20unlikely%20this%20will%20ever%20happen%20and%20even%20if%20it%20does%20you%20wouldn%27t%20really%20know%20what%20to%20do%20about%20it%20anyway.

Reply:
WDYM?

Reply:
if we switch to `@epic-web/invariant`, we get:
> You're fine throwing a descriptive error here

for free, right? the issue with the current `tiny-invariant` package is that it strips out error messages

both packages are just small wrappers around throwing errors anyway, right?

Reply:
Well, possibly, yeah. Tiny invariant strips messages only for kb savings: 

![CleanShot 2025-01-07 at 17 04 59@2x](https://github.com/user-attachments/assets/a9d74479-ea46-4cf7-90c5-5967dff75b63)

I think for us it's not a big deal...